### ISSUE TITLE
Bug downloading files from clients with replicas with clients

### ISSUE THREAD

**Issue by freddyaboulton** [gradio-app/gradio#5263]
### Describe the bug

As noted [here](https://huggingface.slack.com/archives/C02QZLG8GMN/p1692345753750099), when connecting to a space that has replicas under the hood, the client may fail to download temp files associated with a prediction. This is because the file is located on a specific replica and the load balancer may not put you in the correct replica.

@XciD can help us get the url of the replica the file is in.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

-

### Screenshot

-

### Logs

```shell
-
```

### System Info

```shell
-
```

### Severity

I can work around it

**Comment by XciD** [comment]
We have two solutions:
1. Use two env variables `HOSTNAME` and `SPACE_HOST`
```
HOSTNAME: s-xcid-test-adrien-4-9007e-5b6bcfbd89-r2z6g
SPACE_HOST: https://xcid-test-adrien-4.hf.space/
```
Do some magic to obtain:
```
https://xcid-test-adrien-4--r2z6g.hf.space/
```


2. Add an header in the proxy with the direct url `X-Direct-Url` (for example), and re-use it

we can’t compute the env var before spawning the pod as the name at the end is random and only known at runtime
second solution is 1 line of code in proxy code, so quite easy

**Comment by abidlabs** [comment]
Side note: we should confirm that embedding demos with multiple replicas works after this fix.

**Comment by abidlabs** [comment]
I'll take this issue if you haven't already started on it @freddyaboulton.

**Comment by freddyaboulton** [comment]
I did start. Let me push up so you can work off it.

**Comment by abidlabs** [comment]
Closed by #5668
